onchain/rollups/test/foundry/consensus/quorum/Quorum.t.sol

```diff
@@ -0,0 +1,270 @@
+        assertEq(quorum.validators(), validators);
+        assertEq(quorum.totalShares(), sum(shares));
+        assertEq(address(quorum.getHistory()), address(history));
+
+        for (uint256 i; i < numOfValidators; ++i) {
+            address validator = validators[i];
+            uint256 id = quorum.validatorId(validator);
+
+            assertEq(quorum.payee(i), validator);
+            assertEq(quorum.shares(validator), shares[i]);
+            assertEq(quorum.validatorById(id), validator);
+        }
+    }
+
+    function testZeroValidatorId(uint8 numOfValidators, address addr) external {
+        Quorum quorum = deployQuorum(numOfValidators);
+        assertEq(
+            contains(quorum.validators(), addr),
+            quorum.validatorId(addr) != 0
+        );
+    }
+
+    function testValidatorByInvalidId(
+        uint8 numOfValidators,
+        uint256 validatorId
+    ) external {
+        vm.assume(validatorId < 1 || validatorId > numOfValidators);
+        Quorum quorum = deployQuorum(numOfValidators);
+        assertEq(quorum.validatorById(validatorId), address(0));
+    }
+
+    function testSubmitClaimRevertsNotValidator(
+        uint8 numOfValidators,
+        address caller,
+        bytes calldata claim
+    ) external {
+        Quorum quorum = deployQuorum(numOfValidators);
+        vm.assume(quorum.validatorId(caller) == 0);
+        vm.prank(caller);
+        vm.expectRevert("Quorum: sender is not validator");
+        quorum.submitClaim(claim);
+    }
+
+    function testNumOfValidatorsInFavorOfClaim(
+        uint8 numOfValidators,
+        bytes calldata claim
+    ) external {
+        Quorum quorum = deployQuorum(numOfValidators);
+        assertEq(quorum.numOfValidatorsInFavorOf(claim), 0);
+    }
+
+    function testIsValidatorInFavorOf(
+        uint8 numOfValidators,
+        uint256 validatorId,
+        bytes memory claim
+    ) external {
+        Quorum quorum = deployQuorum(numOfValidators);
+        assertFalse(quorum.isValidatorInFavorOf(validatorId, claim));
+    }
+
+    function testValidatorsInFavorOfClaim(
+        uint8 numOfValidators,
+        bytes calldata claim
+    ) external {
+        Quorum quorum = deployQuorum(numOfValidators);
+        assertEq(quorum.validatorsInFavorOf(claim), new address[](0));
+    }
+
+    function testSubmitClaim(bytes memory claim) external {
+        Quorum quorum = deployQuorum(3);
+        bool[] memory submitLog = new bool[](4);
+
+        submitClaim(quorum, claim, submitLog, 1);
+        assertEq(history.numOfClaims(), 0);
+
+        // resubmitting makes no difference
+        submitClaim(quorum, claim, submitLog, 1);
+        assertEq(history.numOfClaims(), 0);
+
+        submitClaim(quorum, claim, submitLog, 2);
+        assertEq(history.numOfClaims(), 1);
+        assertEq(history.claim(0), claim);
+
+        submitClaim(quorum, claim, submitLog, 3);
+        assertEq(history.numOfClaims(), 1);
```

Comment:
we could make this into a `for` loop if we wanna test some quorums with larger size. Because BitMaps uses a new storage slot after, in our case, the first 255 bit positions being used. So quorum with size at least 256 would be ideal.

Reply:
Good point. Let me add a test for 256 validators. :-)